### ISSUE TITLE
Difference is invalid/misleading when comparing a 'null' JSON value and a nested JSON value (array / object)

### ISSUE THREAD

**Issue by brandonwang15** [nsf/jsondiff#21]
### Toy code:
Run in playground: https://go.dev/play/p/XGfPuf_Px8a
```
	j := jsondiff.DefaultJSONOptions()
	a := `{"foo":null}`
	b := `{"foo":["bar"]}`

	_, diff := jsondiff.Compare([]byte(a), []byte(b), &j)
	fmt.Println("comparing null to array is broken")
	fmt.Println(diff)

	a = `{"foo":null}`
	b = `{"foo":{"value": "bar"}}`

	_, diff = jsondiff.Compare([]byte(a), []byte(b), &j)
	fmt.Println("comparing null to object is broken")
	fmt.Println(diff)

	a = `{"foo":null}`
	b = `{"foo":"bar"}`

	_, diff = jsondiff.Compare([]byte(a), []byte(b), &j)
	fmt.Println("comparing null to a non-nested value works!")
	fmt.Println(diff)
```

```
comparing null to array is broken
{
    "foo": {"changed":[null, []]}
}
comparing null to object is broken
{
    "foo": {"changed":[null, {}]}
}
comparing null to a non-nested value works!
{
    "foo": {"changed":[null, "bar"]}
}
```

If a field has a json 'null' value in either of the JSON objects, and the other value is a nested value (array or object), the comparison fails to print out the correct value for the other object field's non-null value. See the playground example above for an illustration.

One way to hack a fix is convert all nil slice values to empty slice values in the struct before serializing to json.

**Comment by nsf** [comment]
Yes, I agree with that this behaviour may not be ideal. But that's how it is. To be honest this lib is a very low priority for me and it's unlikely that I'm going to do anything about it anytime soon.

I can see you're using JSON config. This lib was never meant to provide a machine readable output. I don't even recall who contributed JSON config option and I never used it myself and not planning to. When printing out something like "null <= {}" the lib means that values are of different type and further comparison makes no sense. Same applies to arrays vs other values. The value is different already, thus no need to go deeper. That's what I had in mind when implementing it.

Maybe one day will do something about it (unlikely soon).
